process/block/preprocess/transactions.go

```diff
@@ -901,6 +914,12 @@ func (txs *transactions) createAndProcessMiniBlocksFromMe(
 		totalTimeUsedForComputeGasConsumed += elapsedTime
 		if err != nil {
 			log.Trace("createAndProcessMiniBlocksFromMe.computeGasConsumed", "error", err)
+			isTxTargetedForDeletion := errors.Is(err, process.ErrMaxGasLimitPerOneTxInReceiverShardIsReached)
```

Comment:
we could add this at interceptor level as well.

Reply:
WIll be added in another PR. I already talked with @iulianpascalau about this earlier protection mechanism